### PR TITLE
EES-1162 Correct validation of unique slug for amendments.

### PR DESCRIPTION
**Pull request by shakes90**

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs [hunk 1]
```diff
@@ -434,14 +435,23 @@ public IEnumerable<Guid> GetReferencedReleaseFileVersions(Guid releaseId, params
         private async Task<Either<ActionResult, bool>> ValidateReleaseSlugUniqueToPublication(string slug,
             Guid publicationId, Guid? releaseId = null)
         {
-            if (await _context.Releases.AnyAsync(r => r.Slug == slug && r.PublicationId == publicationId && r.Id != releaseId))
+            var releases = await _context.Releases
+                .Include(r => r.Publication)
+                .Where(r => r.PublicationId == publicationId).ToListAsync();
+             
+            if (releases.Any(release => release.Slug == slug && release.Id != releaseId && IsLatestVersionOfRelease(releases, release.Id)))
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:442]: This condition confuses me (and seems inefficient given the second `Any()` call). Given the above comment about filtering out release on `releaseId`, maybe we could simplify this to:

```csharp
if (releases.Any(release => release.Slug == slug && release.PreviousVersionId != releaseId))
```

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs [hunk 2]
```diff
@@ -434,14 +435,23 @@ public IEnumerable<Guid> GetReferencedReleaseFileVersions(Guid releaseId, params
         private async Task<Either<ActionResult, bool>> ValidateReleaseSlugUniqueToPublication(string slug,
             Guid publicationId, Guid? releaseId = null)
         {
-            if (await _context.Releases.AnyAsync(r => r.Slug == slug && r.PublicationId == publicationId && r.Id != releaseId))
+            var releases = await _context.Releases
+                .Include(r => r.Publication)
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:439]: Do we need to include the publication if we're not using it?

Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs]: Yeah good spot - Thanks - Done

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs [hunk 3]
```diff
@@ -434,14 +435,23 @@ public IEnumerable<Guid> GetReferencedReleaseFileVersions(Guid releaseId, params
...
+                .Where(r => r.PublicationId == publicationId).ToListAsync();
```
[the remark this answers is not in the document]
Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs]: No it needs to be as is. If you add the exclusion of the current release into the initial clause then if that Version were an amendment then it's originating version(s) would be picked up & then latest version of the release would be incorrectly determined.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs (hunk 1 above)
Comment (shakes90) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs:440]: as previous comment.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs]: Cool, think I understand your rationale now thanks!

**Pull request merged**